pretty sure this is not my fault:
```
/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/Output/swift-build-self-host.py.tmp.dir/swiftpm/Sources/SPMLLBuild/llbuild.swift:15:19: error: no such module 'llbuild'
18:39:47 @_exported import llbuild
18:39:47                   ^
18:39:47 error: terminated(1): /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/buildbot_linux/none-swift_package_sandbox_linux-x86_64/usr/bin/swift-build-tool -f /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/buildbot_linux/none-swift_package_sandbox_linux-x86_64/tests/Output/swift-build-self-host.py.tmp.dir/swiftpm/.build/debug.yaml main output:
```

@ddunbar is this a known issue?

--- a/include/swift/Reflection/ReflectionContext.h
+++ b/include/swift/Reflection/ReflectionContext.h
@@ -189,8 +189,27 @@ public:
         LocalStartAddress,
         RemoteStartAddress};
 
-    savedBuffers.push_back(std::move(Buf));
     this->addReflectionInfo(info);
+
+    // Find the __DATA segment.
+    for (unsigned I = 0; I < NumCommands; ++I) {
+      auto CmdBuf = this->getReader().readBytes(
+          RemoteAddress(CmdStartAddress.getAddressData() + Offset),
+          SegmentCmdHdrSize);
+      auto CmdHdr = reinterpret_cast<typename T::SegmentCmd *>(CmdBuf.get());
+      if (strncmp(CmdHdr->segname, "__DATA", sizeof(CmdHdr->segname)) == 0) {
+        auto DataSegmentEnd =
+            ImageStart.getAddressData() + CmdHdr->vmaddr + CmdHdr->vmsize;
+        assert(DataSegmentEnd > ImageStart.getAddressData() &&
+               "invalid range for __DATA");
+        imageRanges.push_back(
+            std::make_tuple(ImageStart, RemoteAddress(DataSegmentEnd)));
+        break;
+      }
+      Offset += CmdHdr->cmdsize;
+    }
+
+    savedBuffers.push_back(std::move(Buf));
     return true;
   }
 

